app/helpers/companies_helper.rb

```diff
@@ -21,10 +21,33 @@ def full_uri company
   end
 
 
+  # Given a collection of companies, create an array of {latitude, longitude, marker}
+  # for each company.  (Can be used by javascript to display markers for many companies)
+  # if link_name is true (the default), the name of each company should be a link to its page
+  #  else the name of each company should just be the name with no link to it
+  def location_and_markers_for(companies, link_name: true)
+
+    results = []
+
+    companies.each do |company|
+      link_name ? name_html = nil : name_html = company.name
+      results << {latitude: company.main_address.latitude,
+                  longitude: company.main_address.longitude,
+                  text: html_marker_text(company, name_html: name_html) }
+    end
+
+    results
+  end
```

Comment:
If some of the companies in `companies` collection lack geolocation, do we want to perform lazy lookup?  I am testing with 4 companies but only two have lat/long coordinates.